### ISSUE TITLE
[Question] Train in Headless Mode/ directly in Python

### ISSUE THREAD

**Issue by hno2** [ilastik/ilastik#1999]
Hey there, 

I'm very stuck here, so I hope you can point me in to the right start/direction. 

## UseCase
I want to train ilastik pixel classification by labeling images in a web frontend via freedrawing. 
I get arrays of the label markings and the image file. 
## Question 
**How would I start using ilastik as a backend for my system?**
## My Guess
I looked at [opPixelClassification.py](../blob/master/ilastik/applets/pixelClassification/opPixelClassification.py), but wasn't getting very var.

Thanks in Advance for your help

**Comment by k-dominik** [comment]
In order to train with your own annotations you basically need to have a "full ilastik" running. Meaning you'd have to have the full computational graph with the operators for reading, feature computation, random forest training, prediction... This is currently not possible in an easy way, but I can assure you that we're working on that. So right now I'd say it is very involved, but it might be a good idea to let the plan rest for 2, 3 months and then get back to us, again.

**Comment by hno2** [comment]
Thanks for your answer @k-dominik I found [train_headless.py](https://github.com/ilastik/ilastik/blob/master/bin/train_headless.py) but I couldn't get it to work first try for the following reasons 

- [ ] [parse_known_cmdline_args() ](https://github.com/ilastik/ilastik/blob/1579e869148edcca2944ca4744333a57f20a4721/bin/train_headless.py#L150) is not working because bbac5e5c21da19f07a6373d6a8cf7c48317133e4 is somehow not included in both the 1.3.2post1 and 1.3.3b1 release I downloaded from the website eventhough it was merged to master in Decemeber.
- [ ] The Example  in the Code `./ilastik-1.1.7-Linux/bin/python train_headless.py MyNewProject.ilp /tmp/cell-slide.png cell-labels.png` is not working, because of path errors for the Project file. In my case I needed to specify an absolute path for the project file.  `./ilastik-1.1.7-Linux/bin/python train_headless.py ./MyNewProject.ilp /tmp/cell-slide.png cell-labels.png`

**Comment by jjerphan** [comment]
Hi @k-dominik and @hno2,

I would like to train classifiers in headless mode with auto-context on several examples with their labels.
I've tried using the [`train_headless.py`](https://github.com/ilastik/ilastik/blob/master/bin/train_headless.py) script too, but I got an error because `PixelClassificationWorkflow.ROLE_NAMES` is not defined any more (at least on Ilastik 1.3.3post3).

Would it be interesting to have it updated and adapted for autocontext too? If so, I can invest some time on such a contribution. :slightly_smiling_face:
